File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
n = int(input())
pointer1 = 1
pointer2 = 1
count = 1
sum = 1

while pointer2 != n:
    if sum == n:
        count += 1
        pointer2 += 1
        sum += pointer2
    elif sum > n:
        sum -= pointer1
        pointer1 += 1
    else:
        pointer2 += 1
        sum += pointer2

print(count)

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.util.*;

public class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int count = 1, start_idx = 1, end_index = 1, sum = 1;
        while (end_index != N) {
            if (sum == N) {
                count++;
                end_index++;
                sum += end_index;
            } else if (sum > N) {
                sum -= start_idx;
                start_idx++;
            } else if (sum < N) {
                end_index++;
                sum += end_index;
            }
        }
        System.out.println(count);
    }
}